code/algorithms.pyx

```diff
@@ -0,0 +1,380 @@
+    
+    Parameters:
+    graph_generator: Function which returns a graph as an adjacency list. This adjacency list should be a numpy array of shape (N,N+1) in which row i stores the indices of the neighbors of node i followed by -1 values until the end of the row.
+    graph_generator_arguments: List of arguments passed to the graph_generator function.
+    w:  Selection strength.
+    c:  Costs.
+    n_graphs: The number of graphs generated with graph_generator.
+    n_runs: The number of simulations on each graph.
+    benefit_to_cost_ratios: List of b/c ratios.
+    dview (optional): ipyparallel.client.view.DirectView object for parallelization. The default value is None, meaning no parallelization.
+    iteration_type (optional): Either 1 (death-birth process) or 0 (imitation process). Default is 1. 
+    
+    Returns:
+    Fixation probabilities as a numpy-array of the same length as benefit_to_cost_ratios.
+    '''
+    fixation_probabilities = []
+    seeds = [rand() for i in xrange(n_graphs)]
+    for bc_ratio in benefit_to_cost_ratios:
+        b = c * bc_ratio
+        if dview is None:
+            results = [measure_fixation_probability(graph_generator(*graph_generator_arguments), n_runs, b, c, w, s, iteration_type) for s in seeds]
+        else:
+            dview.push(dict(graph_generator=graph_generator, graph_generator_arguments=graph_generator_arguments, n_runs=n_runs, b=b, c=c, w=w, iteration_type=iteration_type))
+            dview.scatter('seeds',seeds)
+            dview.execute('results = [measure_fixation_probability(graph_generator(*graph_generator_arguments), n_runs, b, c, w, s, iteration_type) for s in seeds]')
+            results = dview.gather('results')
+        fixation_probabilities.append(np.mean(results))
+    return np.array(fixation_probabilities) 
+
+
+def measure_fixation_probability(np.ndarray[np.int64_t, ndim=2] graph_nd, int n_runs, float b, float c, float w, int seed, int iteration_type):
+    '''Measure fixation probability by running many simulations starting with a random node as the initial cooperator. 
+    
+    Parameters:
+    graph_nd: The graph given as a numpy array of shape (N,N+1) in which row i stores the indices of the neighbors of node i followed by -1 values until the end of the row.
+    n_runs: Number of simulations.
+    b: Benefit value.
+    c: Cost value.
+    w: Selection strength.
+    seed: Seed for the random number generator.
+    iteration_type: Either 1 (death-birth process) or 0 (imitation process).
+    
+    Returns:
+    The ratio of the number of simulations that ended with a fixation of cooperators.
+    '''
+    srand(seed)
+    cdef long int [:,:] graph = graph_nd
+    cdef int N = graph.shape[0]
+    cdef int first_cooperator_index
+    cdef int C_fixations = 0
+    cdef int i = 0
+    for i in range(n_runs):
+        first_cooperator_index = int((rand() / (float(INT_MAX) + 1) * N))
```

Comment:
I do not expect this to be a problem for the paper, but I thought it is worth of a comment that `rand()` is depreciated. There are other (more elegant) ways how to generate random numbers in C++ (I am not entirely sure how it would work in cython).

I found extremly useful this lecture
https://www.youtube.com/watch?v=LDPMpc-ENqY

Using [Mersenne Twister](https://en.wikipedia.org/wiki/Mersenne_Twister) algorithm for random numbers would also ensure seed-based reproducibility on different platforms / compilers.